### DISCUSSION TITLE
Merging Ground Truths in Different Layers

### DISCUSSION THREAD

**Discussion by davidteeters** [UB-Mannheim/escriptorium#28] [Q&A]
Hello! I am a graduate student at LSE using eScriptorium on 18th century financial ledgers. 

Two questions:

1. I have ground truths (that I have manually labeled) in different layers of eScriptorium. I have tried to export the two sets and merge the METS files before reimporting but so far have been unsuccessful. Any suggestions on how to approach this please?

2. Many of the individual lines of text have additional sub-lines added in above / below (indicated by carets). Has anyone else run into this problem, found a solution?

Thank you,

David

**Comment by stweil** [comment]
1. I'd ignore the METS files and only merge the PAGE or ALTO files with the desired text layers. Then create a ZIP file with these files and import it.
2. I am not sure that I have understood your question correctly, but it sound as if you have an overlay of several segmentations. This results in more than one baseline per text line. If this is the case, you could delete the superfluous baselines. Or start with a new segmentation and overwrite the existing one. Or import XML files with a correct segmentation and overwrite the existing one.

**Reply by davidteeters** [reply]
Thank you stweil for the prompt reply! Problem one resolved.

Here's a screenshot of what I mean for the second problem, with a sub-line of text included below but in the mask of the main line (but not picked up by the trained model). Do I need to have a separate baseline for each, and then find a way to link them later? 

For this model (fine tuned off of a public model in kraken) I have approximately 150 images (each with 50-100 lines) of manually labeled ground truths, but I'm stuck at around 82% accuracy. I'm hoping to get it >90%, don't know whether best to invest the time into more manual labeling or to try different models (TrOCR?). 

Any suggestions / experience much appreciated!

<img width="1023" alt="Screenshot 2025-06-10 at 9 05 49 PM" src="https://github.com/user-attachments/assets/5c53a976-a59c-46c0-8a79-035dcfb1250d" />

**Reply by stweil** [reply]
Yes, I am afraid that the neural network will not be able to combine line and sub-line into a single text line.

Both lines need separate baselines, and the line masks should only include single lines. As you said this means that you have to find a way to link them later.

**Comment by davidteeters** [comment]
This is great, thank you! I'll go back and amend the ground truths to reflect this.

Another question: can the model differentiate between color of ink used in the original ledgers? There are reference numbers written in red, which contains informational value, and so I've indicated them with the unique character identifier of [brackets].

As my base model I'm using the open source McCATMuS_nfd_nofix_V1, which has a ~93% accuracy. Despite fine tuning with 150 images, I'm still stuck at 83% accuracy. In your experience, how many ground truths are needed to fine tune a model like this? I'm trying to determine where I can best prioritize the allocation of my time.

Best,

David

**Reply by stweil** [reply]
No, the current models for kraken (which is used by eScriptorium for the text recognition) are not trained to recognize the text color. This and also the missing ability to recognize text attributes like bold, italic, underlined (...) is a major deficit of current OCR software. Large Language Models can recognize text, color and attributes. Therefore I expect that such LLMs will become more and more important for special text recognition tasks. They are still too expensive for mass production of OCR.

**Reply by stweil** [reply]
Typically I use a small number of page images for fine tuning, 5–20 pages for printed text (based on our _german_print_ model), 10–50 for handwritten text (based on our _german_handwriting_).
